portfolio/src/main/webapp/js/comments.js

```diff
@@ -143,28 +145,30 @@ function postComment(event) {
 /**
  * Creates a function that sends a comment delete request.
  * @param {number} id
+ * @param {Element} commentElement
  * @return Returns a function that deletes the comment associated with an ID.
  */
-function createCommentDeleter(id) {
+function createCommentDeleter(id, commentElement) {
   return () => {
     const deleteUrl = 
         new URL("/comments", `${location.protocol}//${location.hostname}`);
     deleteUrl.searchParams.set("id", id);
 
     fetch(deleteUrl, { method: 'DELETE' }).then((resp) => {
-        if (resp.ok) {
-          addNotification("Comment deleted successfully.", "alert-success");
-          loadComments();
-        } else {
-          return resp.text().then(
-              text => Promise.reject(`Error ${resp.status}: ${text}`));
-        }
```

Comment:
I'm curious, why didn't removing the comment from the datastore then reloading the page not work? Did the DELETE method return before the comment is deleted from the datastore?

Reply:
Yep. Datastore has a bit of lag on input/deletion.

Reply:
Yeah I suspect it’s either some sort of caching or asynchronous operation going on. Although Joe that was definitely my thought process as well, hence running into the issue!